src/components/settings/namespaces/NamespaceListbox.tsx

```diff
@@ -13,7 +14,26 @@ type NamespaceLisboxProps = {
 export default function NamespaceListbox(props: NamespaceLisboxProps) {
   const { disabled, namespaces, className } = props;
 
-  const { currentNamespace, setCurrentNamespace } = useNamespace();
+  const { currentNamespace } = useNamespace();
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const setCurrentNamespace = (namespace: SelectableNamespace) => {
+    // navigate to the current location.path with the new namespace prependend
+    // e.g. /namespaces/default/segments -> /namespaces/namespaceKey/segments
+
+    const path = location.pathname.split('/');
+
+    // if path does not begin with namespaces/:namespaceKey, prepend it
+    if (path[1] !== 'namespaces') {
+      path[1] = 'namespaces';
+    }
```

Comment:
yeah.. i want to add some specific tests for this functionality as it could be a bit hairy. lemme see what I can come up with and if I can fix the `segments` -> `flags` experience

Reply:
I updated the code to extract the namespace/path stuff to a function and added some unit tests:

https://github.com/flipt-io/flipt-ui/pull/188/files#diff-d4a40b1fc30edde55851771fd299c6eac392bd4d1afed874c073926c96ebd664
https://github.com/flipt-io/flipt-ui/pull/188/files#diff-d9b0865f2e8f841410cd4faf3e1e4a1fb7cb0b545f583f6adaad5841d7eaa58d